category_encoders/target_encoder.py

```diff
@@ -113,34 +127,52 @@ def _fit(self, X, y, **kwargs):
 
     def fit_target_encoding(self, X, y):
         mapping = {}
+        prior = self._mean = y.mean()
 
         for switch in self.ordinal_encoder.category_mapping:
             col = switch.get('col')
-            values = switch.get('mapping')
+            if 'HEIR_' not in col:
+                values = switch.get('mapping')
 
-            prior = self._mean = y.mean()
+                scalar = prior
+                if self.heirarchy and col in self.heirarchy:
+                    col_heir = 'HEIR_'+col
+                    stats_heir = y.groupby(X[col_heir]).agg(['count', 'mean'])
+                    smoove_heir = self._weighting(stats_heir['count'])
+                    scalar_heir = scalar * (1 - smoove_heir) + stats_heir['mean'] * smoove_heir
```

Comment:
I would find this more readable if you write `prior` instead of `scalar`. This should be the same right?

Reply:
No, I don't believe that is true.  The prior is taken over all the data, but the scalar is an estimate of the probability.  The paper clearly defines it separately, so feel the language should follow.

Reply:
The variable called `scalar_heir` is defined separately and should keep its name. My comment war referring to the variable called `scalar` on the same line. This is even set equal to `prior` in line 155.

Reply:
So, the scalar is normally set to the prior, but for multiple hierarchies, the scalar is a function.  This matters more when we will have multi-level hierarchies, in which the "scalar" will definitely move away from the prior.  So, in that sense, I think it matters that we don't call it the prior.  The prior and scalar will be different and that also matters for the default values (-1 and -2) in the mapping.

Reply:
you're right! thanks for that explanation